src/feature/relay/dns.c

```diff
@@ -1872,16 +1878,22 @@ evdns_wildcard_check_callback(int result, char type, int count, int ttl,
       for (i = 0; i < count; ++i) {
         char answer_buf[INET_NTOA_BUF_LEN+1];
         struct in_addr in;
+        int ntoa_res;
         in.s_addr = addrs[i];
-        tor_inet_ntoa(&in, answer_buf, sizeof(answer_buf));
-        wildcard_increment_answer(answer_buf);
+        ntoa_res = tor_inet_ntoa(&in, answer_buf, sizeof(answer_buf));
+        tor_assert_nonfatal(ntoa_res > 0);
```

Comment:
This is incorrect; look at the documentation for tor_inet_ntoa().  It says "returns -1 on failure" and "returns non-negative integer on success."  So the test should be `>= 0`.

Reply:
All three check for `tor_inet_ntoa()` return value fixed.